asab/library/providers/git.py

```diff
@@ -0,0 +1,127 @@
+import tempfile
+import logging
+import os
+import shutil
+
+import pygit2
+
+from .filesystem import FileSystemLibraryProvider
+from ...config import Config
+
+#
+
+L = logging.getLogger(__name__)
+
+#
+
+
+class GitLibraryProvider(FileSystemLibraryProvider):
+	"""
+	Read-only git provider to read from remote repository.
+	It clones a remote git repository to a temporary directory and then uses the
+	FileSystemLibraryProvider to read the files.
+	To read from local git repository, please use FileSystemProvider.
+
+	Configuration:
+	(Use either deploytoken, publickey+privatekey for SSH option, or username and password and HTTP access.)
+	```
+	[library]
+	providers=git+<URL or deploy token>
+
+	[library.GitProvider]
```

Comment:
`[library:git]` ... "Provider" is an implementation detail, we don't require users to learn that.